### ISSUE TITLE
Migrate flutter_adapter.dart and flutter_test_adapter.dart to use super params

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#101455]
In a previous attempt this was breaking in google3:

```
google3:///[third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart:27](https://cs.corp.google.com/piper///depot/google3/third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart?l=27&ws=tap-prod-presubmit/179932707&snapshot=7):11: Error: The parameter 'ipv6' can't have a value of 'null' because of its type 'bool', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    super.ipv6,
          ^^^^
google3:///[third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart:28](https://cs.corp.google.com/piper///depot/google3/third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart?l=28&ws=tap-prod-presubmit/179932707&snapshot=7):11: Error: The parameter 'enableDds' can't have a value of 'null' because of its type 'bool', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    super.enableDds,
          ^^^^^^^^^
google3:///[third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart:29](https://cs.corp.google.com/piper///depot/google3/third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart?l=29&ws=tap-prod-presubmit/179932707&snapshot=7):11: Error: The parameter 'enableAuthCodes' can't have a value of 'null' because of its type 'bool', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    super.enableAuthCodes,
          ^^^^^^^^^^^^^^^
google3:///[third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart:27](https://cs.corp.google.com/piper///depot/google3/third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart?l=27&ws=tap-prod-presubmit/179932707&snapshot=7):11: Error: The parameter 'ipv6' can't have a value of 'null' because of its type 'bool', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    super.ipv6,
          ^^^^
google3:///[third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart:28](https://cs.corp.google.com/piper///depot/google3/third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart?l=28&ws=tap-prod-presubmit/179932707&snapshot=7):11: Error: The parameter 'enableDds' can't have a value of 'null' because of its type 'bool', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    super.enableDds,
          ^^^^^^^^^
google3:///[third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart:29](https://cs.corp.google.com/piper///depot/google3/third_party/dart/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart?l=29&ws=tap-prod-presubmit/179932707&snapshot=7):11: Error: The parameter 'enableAuthCodes' can't have a value of 'null' because of its type 'bool', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    super.enableAuthCodes,
          ^^^^^^^^^^^^^^^
```

There were no issues in open-source land, though. We need to investigate why this wasn't working in google3.

**Comment by christopherfujino** [comment]
@goderbauer Do you have a link to a CL?

**Comment by goderbauer** [comment]
cl/439752165 is a roll CL where we saw the failures.

This is were we reverted things because of the g3 failures: https://github.com/flutter/flutter/pull/101436

**Comment by goderbauer** [comment]
I wonder if this is related to https://github.com/dart-lang/sdk/issues/48769.

**Comment by christopherfujino** [comment]
> dart-lang/sdk#48769

Ahh, that would make sense

**Comment by christopherfujino** [comment]
> > [dart-lang/sdk#48769](https://github.com/dart-lang/sdk/issues/48769)
> 
> Ahh, that would make sense. So I'm guessing right now you would have to copy the default value from the super constructor?

Although that still doesn't explain why this compiled on GitHub...

**Comment by goderbauer** [comment]
> So I'm guessing right now you would have to copy the default value from the super constructor?

In some cases, the default value is automatically copied. But it seems like there are some bugs in the dart front end around this (or I am not understanding fully in what cases they can or cannot be copied). However, if there are cases where copying doesn't work, the analyzer should really warn you about it instead of waiting until compile time to figure this out. Let's see what the dart team says on the bug...

**Comment by goderbauer** [comment]
https://dart-review.googlesource.com/c/sdk/+/240844 has been identified as a potential root cause for this. Once that fix has rolled into google, we can migrate these files again to check.

**Comment by christopherfujino** [comment]
> https://dart-review.googlesource.com/c/sdk/+/240844 has been identified as a potential root cause for this. Once that fix has rolled into google, we can migrate these files again to check.

cool

**Comment by goderbauer** [comment]
The fix for the underlying bug landed in Dart 2.18.0-17.0.dev, which has rolled into Flutter now.
